db/collection.go

```diff
@@ -148,66 +149,100 @@ func (c *Collection) AddIndex(config IndexConfig) error {
 // ReadTxn creates an explicit readonly transaction. Any operation
 // that tries to mutate an instance of the collection will ErrReadonlyTx.
 // Provides serializable isolation gurantees.
-func (c *Collection) ReadTxn(f func(txn *Txn) error) error {
-	return c.db.readTxn(c, f)
+func (c *Collection) ReadTxn(f func(txn *Txn) error, opts ...TxnOption) error {
+	return c.db.readTxn(c, f, opts...)
 }
 
 // WriteTxn creates an explicit write transaction. Provides
 // serializable isolation gurantees.
-func (c *Collection) WriteTxn(f func(txn *Txn) error) error {
-	return c.db.writeTxn(c, f)
+func (c *Collection) WriteTxn(f func(txn *Txn) error, opts ...TxnOption) error {
+	return c.db.writeTxn(c, f, opts...)
 }
 
 // FindByID finds an instance by its ID.
 // If doesn't exists returns ErrNotFound.
-func (c *Collection) FindByID(id core.InstanceID) (instance []byte, err error) {
+func (c *Collection) FindByID(id core.InstanceID, opts ...TxnOption) (instance []byte, err error) {
 	_ = c.ReadTxn(func(txn *Txn) error {
 		instance, err = txn.FindByID(id)
 		return err
-	})
+	}, opts...)
 	return
 }
 
-// Create creates instances in the collection.
-func (c *Collection) Create(vs ...[]byte) (ids []core.InstanceID, err error) {
+// Create creates an instance in the collection.
+func (c *Collection) Create(v []byte, opts ...TxnOption) (ids []core.InstanceID, err error) {
```

Comment:
This should just return a single InstanceID

Reply:
Good catch 👍